Quantum-Control-Applications/ESR/pi_pulse_calibration.py

```diff
@@ -0,0 +1,175 @@
+from qm import SimulationConfig
+from qm.qua import *
+from qm import LoopbackInterface
+from qm.QuantumMachinesManager import QuantumMachinesManager
+from configuration import *
+import matplotlib.pyplot as plt
+from qm.simulate.credentials import create_credentials
+
+###################
+# The QUA program #
+###################
+
+pulse1_min = 40 // 4
+pulse1_max = 400 // 4
+dpulse1 = 40 // 4
+
+pulses1 = np.arange(pulse1_min, pulse1_max + 0.1, dpulse1)
+
+cooldown_time = int(10e6 // 4)
+
+n_avg = 100
+
+with program() as pi_pulse_cal:
+
+    n = declare(int)
+    pulse1_len = declare(int)
+    pulse2_len = declare(int)
+    pulse_delay = declare(int)
+    readout_delay = declare(int)
+
+    I = declare(fixed)
+    Q = declare(fixed)
+    I_st = declare_stream()
+    Q_st = declare_stream()
+
+    with for_(n, 0, n < n_avg, n + 1):
+
+        with for_(pulse1_len, pulse1_min, pulse1_len <= pulse1_max, pulse1_len + dpulse1):
+
+            # wait(cooldown_time)
+
+            reset_phase("ensemble")
+            reset_phase("resonator")
+            reset_frame("ensemble")
+
+            echo = declare_stream(adc_trace=True)
+
+            assign(pulse2_len, 2 * pulse1_len)
+            assign(
+                pulse_delay,
+                2000 // 4 - Cast.mul_int_by_fixed(pulse1_len, 0.5) - Cast.mul_int_by_fixed(pulse2_len, 0.5) - 8,
```

Comment:
this `Cast.mul_int_by_fixed(pulse2_len, 0.5)` is actually pulse1_len no?
To get these values, you first rune the simulator right?